lib/graphiti/resource/remote.rb

```diff
@@ -0,0 +1,48 @@
+module Graphiti
+  class Resource
+    module Remote
+      extend ActiveSupport::Concern
+
+      included do
+        self.adapter = Graphiti::Adapters::JSONAPI
+        self.model = OpenStruct
+        self.validate_endpoints = false
+
+        class_attribute :timeout,
+          :open_timeout
+      end
+
+      def save(*args)
+        raise Errors::RemoteWrite.new(self.class)
+      end
+
+      def destroy(*args)
+        raise Errors::RemoteWrite.new(self.class)
+      end
+
+      def before_resolve(scope, query)
+        scope[:params] = Util::RemoteParams.generate(self, query)
+        scope
+      end
+
+      # Forward all headers
+      def request_headers
+        if defined?(Rails)
+          context.request.headers.to_h.reject { |k, v| k.include?('.') }
+        else
+          {}
+        end
+      end
+
+      def make_request(url)
+        headers = request_headers.dup
+        headers['Content-Type'] = 'application/vnd.api+json'
+        Faraday.get(url, nil, headers) do |req|
```

Comment:
Since we're making this an optional dependency, would be nice to require and raise:

```ruby
unless defined?(Faraday)
  raise "Faraday not defined. Please require the 'faraday' gem to use remote resources"
end
```

Reply:
Good call, force-pushed